### PR TITLE
feat: openai migrate types

### PR DESCRIPTION
**Pull request by nsbradford**
https://app.grit.io/studio?key=od7xP6vZs81f2M0uAl_90

### CODE REVIEW

**Comment by nsbradford** [comment]
Current dependencies on/for this PR:
* main
  * **PR #154** <a href="https://app.graphite.dev/github/pr/getgrit/js/154" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/getgrit/js/154?utm_source=stack-comment).

**Review by seren5240 — CHANGES_REQUESTED** [review]
The pattern body is customer-facing. We shouldn't merge TODOs and commented code.

**Comment by nsbradford** [comment]
Removed comments and TODOs, Putting TODOs here, see https://withgrit.slack.com/archives/C051Z9PK5SN/p1691798721024679
    // TODO tried and failed to replace with "and" or use "maybe or"?
    // TODO we ideally want extra type guards so it only changes things that were imported from 'openai'.
    //   e.g. "Model" is a very dangerous thing to be changing, i have disabled for this reason.
    // TODO maybe try using AST nodes, but i found too many edge cases - 
    //   e.g. type_annotation(type="ChatCompletionRequestMessage") => `: foo`

File: .grit/patterns/openai_v4.md [hunk 1]
```diff
@@ -115,14 +115,64 @@ pattern change_completion_try_catch() {
     }
 }
 
+pattern change_imports() {
+    any {
```
Comment (seren5240) [inline .grit/patterns/openai_v4.md:119]: This `any` clause looks redundant.

**Comment by nsbradford** [comment]
see discussion: https://withgrit.slack.com/archives/C051Z9PK5SN/p1691852465422899?thread_ts=1691798721.024679&cid=C051Z9PK5SN

File: .grit/patterns/openai_v4.md (hunk 1 above)
Comment (morgante) [reply .grit/patterns/openai_v4.md:119]: @nsbradford Why did you add an empty any? An any with a single nested entry is useless.

**Review by morgante — CHANGES_REQUESTED** [review]
Remove https://github.com/getgrit/js/pull/154/files/5e4e2265213547b360c04116c61fa19355006d93#diff-47037a0d6a2ed518ebcfcf992da7d99f88ecfaf4e105032d58500b908e50d437R24-R30

**Comment by nsbradford** [comment]
> Remove https://github.com/getgrit/js/pull/154/files/5e4e2265213547b360c04116c61fa19355006d93#diff-47037a0d6a2ed518ebcfcf992da7d99f88ecfaf4e105032d58500b908e50d437R24-R30

@morgante that is still being used, removing it breaks a bunch of things

**Review by morgante — CHANGES_REQUESTED** [review]
Your approach to imports is redundant. Please reconcile with https://github.com/getgrit/js/pull/154/files#diff-47037a0d6a2ed518ebcfcf992da7d99f88ecfaf4e105032d58500b908e50d437R24-R30

We should not be rewriting the import in two places. Your approach is going to fall down as soon as someone has a require statement.

File: .grit/patterns/openai_v4.md [hunk 2]
```diff
@@ -21,13 +21,7 @@ pattern change_constructor() {
...
         } => .,
         $params => `$details`,
-        $program <: contains or {
-            `import $old from $src`,
-            `$old = require($src)`
-        } where {
-            $src <: `"openai"`,
-            $old => `OpenAI`
-        }
+        $program <: contains bubble change_imports(),
```
Comment (morgante) [inline .grit/patterns/openai_v4.md:24]: Pattern calls are auto-bubbled.

```suggestion
        $program <: contains change_imports(),
```

Comment (nsbradford) [reply .grit/patterns/openai_v4.md]: done

**Review by morgante — APPROVED** [review]

**Pull request merged**